### PR TITLE
fix: pinned column resizing in compare view [WEB-1395]

### PR DESCRIPTION
**Pull request by EmilyBonar**
## Description
This is a tricky situation, I'm open to suggestions both on the behavior I chose and how I implemented it. The issue was that on the new experiment list if:

- you had extra columns pinned (beyond just checkboxes and name)
- you opened the comparison pane
- you resized the comparison pane so that it covered the pinned experiments
- and you closed the comparison pane

Then the columns would overlap on the table.

The behavior I implemented to fix this is that when you resize the comparison pane, the difference between the old width and the new width is taken. If the table pane grew, then that difference is added onto the last pinned column. (This is the same behavior as previously) If the table pane shrunk, the last pinned column shrinks until it has exhausted that difference or it reaches a minimum width (set to 40px). If there's any difference left over, it cascades down to the second to last pinned column, and so on. The minimum width for the table pane as a whole is numPinnedCols*40px (+17px to account for scrollbars).

## Test Plan
Follow the steps laid out in the description on Netlify and latest-main and note the differences. Report any oddities in behavior.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Comment by ClaireNeveu** [comment]
When you expand the compare pain, can you divide the added space between all pinned columns instead of just adding to the last one?

**Comment by EmilyBonar** [comment]
Yes, I could make that happen. Would you expect the same sort of behavior in reverse as well, or is the current behavior fine?

**Comment by ClaireNeveu** [comment]
Doing the same is reverse would be ideal.

Also, I notice the columns only resize on dragstop. I realize this might not be possible or performant but did you try resizing them while the drag happens?

**Comment by EmilyBonar** [comment]
Ok, I'll see what I can do. I didn't do resizing during the drag initially due to performance concerns. Trying it out now, it's a little laggy, but I might be able to make it work with a debounce.

**Comment by ClaireNeveu** [comment]
If you can't make it work without being laggy, don't sweat it.

**Review by ClaireNeveu — APPROVED** [review]
Giving you design approval here. The behavior looks really good to me.

I did note at least one bug where columns are still getting overlapped but I'll leave the rest up to engineering review.

**Comment by ClaireNeveu** [comment]
One final design issue: when resizing the compare pane (or any of the columns in compare view), the horizontal scrollbar will show up during part of the drag. You should never see a horizontal scrollbar when the compare view is open.

File: webui/react/src/pages/F_ExpList/F_ExperimentList.tsx [hunk 1]
```diff
@@ -451,16 +451,27 @@ const F_ExperimentList: React.FC<Props> = ({ project }) => {
...
-    (width: number) => {
+    (newTableWidth: number) => {
+      const widthDifference = newTableWidth - comparisonViewTableWidth;
+      // Positive widthDifference: Table pane growing/compare pane shrinking
+      // Negative widthDifference: Table pane shrinking/compare pane growing
+      const newColumnWidths: Record<string, number> = { ...settings.columnWidths };
+      pinnedColumns
+        .filter(
+          (col) =>
+            col !== 'selected' && (widthDifference > 0 || newColumnWidths[col] !== minColumnWidth),
```
Comment (hkang1) [inline webui/react/src/pages/F_ExpList/F_ExperimentList.tsx:462]: **question:** is `selected` the multi-select checkbox column? Maybe make that into a constant of `MULTISELECT` or something so it stands out more

**Review by hkang1 — COMMENTED** [review]
Nice, this works pretty nicely. One area of concern is around how the scrollbar width is hardcoded. My thought is to get it dynamically so it will work better across different browsers since the scrollbar is an area that can vary.

**Review by hkang1 — APPROVED** [review]
LGTM, thanks for the updates. The scrolling issue might be better to be addressed with product design, but the scrollbar issue looks great with the new hook!

**Pull request merged**